src/parser.ts

```diff
@@ -300,9 +300,12 @@ function standaloneName(
   keyNameFromDefinition: string | undefined,
   usedNames: UsedNames
 ): string | undefined {
-  const name = schema.title || schema.id || keyNameFromDefinition
-  if (name) {
-    return generateName(name, usedNames)
+  const nameSources = [schema.title, schema.id, keyNameFromDefinition].filter(
+    source => typeof source !== 'undefined'
+  ) as string[]
```

Comment:
Can you simplify this code a bit by moving the `toSafeString` call into `standaloneName`?

```suggestion
  const name = [schema.title, schema.id, keyNameFromDefinition]
    .filter(Boolean)
    .map(toSafeString)
    .find(_ => _ != null)

  return generateName(name, usedNames)
```